custom_components/flashforge_adventurer_3/sensor.py

```diff
@@ -14,74 +14,91 @@
+    logger.info("Setting up sensor platform for entry %s", config_entry.entry_id)
+    
     config = hass.data[DOMAIN][config_entry.entry_id]
-    if config_entry.options:
-        config.update(config_entry.options)
-    coordinator = FlashforgeAdventurer3Coordinator(hass, config)
+
+    logger.debug("Config: %s", config)
+    
+    coordinator = FlashForgeAdventurer3Coordinator(hass, config)
+    logger.info("Coordinator created")
+    
     await coordinator.async_config_entry_first_refresh()
+    logger.info("First refresh of coordinator done")
+
     sensors = [
-        FlashforgeAdventurer3StateSensor(coordinator, config),
-        FlashforgeAdventurer3ProgressSensor(coordinator, config),
+        FlashForgeAdventurer3StateSensor(coordinator, config),
+        FlashForgeAdventurer3ProgressSensor(coordinator, config),
+        FlashForgeAdventurer5TemperatureSensor (coordinator, config),
     ]
+
+    logger.debug("Sensors list: %s", sensors)
+    
     async_add_entities(sensors, update_before_add=True)
+    logger.info("Added sensor entities")
 
 
-class FlashforgeAdventurer3Coordinator(DataUpdateCoordinator):
+class FlashForgeAdventurer3Coordinator(DataUpdateCoordinator):
     def __init__(self, hass, printer_definition: PrinterDefinition):
         super().__init__(
             hass,
             LOGGER,
             name='My sensor',
             update_interval=timedelta(seconds=60),
         )
-        self.ip = printer_definition['ip_address']
-        self.port = printer_definition['port']
+        self.ip = printer_definition.get('ip_address', 'Unknown IP')  # Set a default IP if not provided
+        self.port = printer_definition.get('port', 8899)  # Set a default port if not provided
+        self.model = printer_definition.get(CONF_MODEL, 'FlashForge Adventurer 3')
+
 
     async def _async_update_data(self):
         async with async_timeout.timeout(5):
-            return await get_print_job_status(self.ip, self.port)
+            return await get_print_job_status(self.ip, self.port, self.model)
 
-
-class FlashforgeAdventurer3CommonPropertiesMixin:
+class FlashForgeAdventurer3CommonPropertiesMixin:
     @property
     def name(self) -> str:
-        return f'FlashForge Adventurer 3'
+        return f'{self.model}'
 
     @property
     def unique_id(self) -> str:
-        return f'flashforge_adventurer_3_{self.ip}'
-
+        return f'{self.model.lower().replace(" ", "_")}_{self.ip}'
 
-class BaseFlashforgeAdventurer3Sensor(FlashforgeAdventurer3CommonPropertiesMixin, CoordinatorEntity, Entity):
+class BaseFlashForgeAdventurer3Sensor(FlashForgeAdventurer3CommonPropertiesMixin, CoordinatorEntity, Entity):
     def __init__(self, coordinator: DataUpdateCoordinator, printer_definition: PrinterDefinition) -> None:
         super().__init__(coordinator)
-        self.ip = printer_definition['ip_address']
+        self.ip = printer_definition['ip_address']  # Use 'ip_address' instead of 'ip'
```

Comment:
What does this comment mean?

Reply:
That's my mistake. I had written it as `['ip']` and left a comment for myself to not make a dumb mistake. I'll remove